File: src/main/java/org/sylfra/idea/plugins/remotesynchronizer/ProjectOpenCloseListener.java
package org.sylfra.idea.plugins.remotesynchronizer;

import com.intellij.openapi.project.Project;
import com.intellij.openapi.project.ProjectManagerListener;
import org.jetbrains.annotations.NotNull;

public class ProjectOpenCloseListener implements ProjectManagerListener {

    @Override
    public void projectClosed(@NotNull Project project) {
        project.getService(FileSyncPlugin.class).projectClosed();
    }
}

File: src/main/kotlin/org/imyuyu/idea/plugins/filesync/ProjectOpenCloseListener.kt
package org.imyuyu.idea.plugins.filesync

import com.intellij.openapi.project.Project
import com.intellij.openapi.project.ProjectManagerListener

class ProjectOpenCloseListener : ProjectManagerListener {
    override fun projectClosed(project: Project) {
        FileSyncPlugin.getInstance(project).projectClosed()
    }
}
